### DISCUSSION TITLE
Focus always on the first field of the first array. (multiple arrays)

### DISCUSSION THREAD

**Discussion by 6vvvvvv** [react-hook-form/react-hook-form#10310] [Q&A] [answered]
Hello,
I'm using usefieldarray and have some arrays rendered in the form. But if I click ADD button**(not the first one)** and the focus will always stay on the first element of the first array.  Does somebody have the idea about this?

Data structure
![image](https://user-images.githubusercontent.com/47306004/233324976-5cc16246-d814-4e15-b879-7cd2c79d966d.png)

Form view
![image](https://user-images.githubusercontent.com/47306004/233325068-14cad2ed-edc7-42a0-a367-16b0d2759383.png)

Related code
![image](https://user-images.githubusercontent.com/47306004/233325502-08b28503-04b6-47fd-8cf3-bc82ae516f11.png)

Thanks in advance

**Comment by leapful** [comment, marked as answer]
@6vvvvvv 

Based on your description, the root cause might be from option `shouldFocusError` which is enabled by default to help focusing on first field element that has error. When appending new field array item, your form might have validation rule to check required field or something like that and triggers focus on error.

Please try to disable the option to see if it works. If not, please prepare a minimal re-producible codesanbox url to help investigate the problem.

**Reply by bluebill1049** [reply]
you can also do the following

`append({...}, focusOptions)` focus options allow you to customise the focus input name

**Reply by 6vvvvvv** [reply]
That's hilarious! I'm unable to replicate the issue in CSB, and the minimal demo seems to be functioning properly. I'll investigate further on my local environment and get back to you all later.

Thanks for your replies, thank you all.

**Reply by 6vvvvvv** [reply]
I have fixed it by removing some codes. Thank you @leapful @bluebill1049
